src/classes/Game.js

```diff
@@ -0,0 +1,137 @@
+/* eslint-disable import/no-unresolved */
+/* eslint-disable no-plusplus */
+/* eslint-disable class-methods-use-this */
+/* eslint-disable prefer-destructuring */
+/* eslint-disable no-restricted-globals */
+/* eslint-disable import/extensions */
+import { DateTime } from 'https://cdn.skypack.dev/luxon';
+import { Snake } from './Snake.js';
+import { Input } from '../../utils/Input.js';
+import { SNAKE_SPEED } from '../../utils/constants.js';
+import { Cherry } from './Cherry.js';
+import { Banana } from './Banana.js';
+
+// eslint-disable-next-line import/prefer-default-export
+export class Game {
+  constructor() {
+    this.lastRenderTime = 0;
+    this.isGameOver = false;
+    this.gameBoard = document.getElementById('game-board');
+    this.snake = new Snake();
+    this.input = new Input();
+    this.food = this.getRandomFoodType();
+    this.snakeSpeed = SNAKE_SPEED;
+  }
+
+  getRandomFoodType() {
+    const food = Math.random() < 0.7 ? new Cherry(this.snake) : new Banana(this.snake);
+    return food;
+  }
+
+  main(currentTime) {
+    if (this.isGameOver) {
+      const username = this.username;
+      const points = this.snake.points;
+      const time = this.time;
+      this.saveScore(username, points, time);
+
+      if (confirm(`${username} lost. Points: ${points} Click OK to restart. ${time}`)) {
+        window.location = '';
+      }
+      return;
+    }
+
+    window.requestAnimationFrame(this.main.bind(this));
+    const secondsSinceLastRender = (currentTime - this.lastRenderTime) / 1000;
+    if (secondsSinceLastRender < 1 / this.snakeSpeed) return;
+
+    this.lastRenderTime = currentTime;
+
+    this.update();
+    this.render();
+  }
+
+  update() {
+    const inputDirection = this.input.getInputDirection();
+    this.snake.movement(inputDirection);
+
+    if (this.snake.onSnake(this.food.food)) {
+      this.snake.grow(this.food);
+      this.food = this.getRandomFoodType();
+      this.snakeSpeed += 0.5;
+    }
+
+    this.checkForGameOver();
+  }
+
+  render() {
+    this.gameBoard.innerHTML = '';
+    this.snake.render(this.gameBoard);
+    this.food.render(this.gameBoard, this.food.frType);
+  }
+
+  checkForGameOver() {
+    this.isGameOver = this.snake.isIntersecting();
+  }
+
+  start() {
+    let username = prompt('Enter username: ');
+
+    while (!username) {
+      username = prompt('Username cannot be null. Please enter a username: ');
+    }
+    this.username = username;
+
+    const currentStartTime = DateTime.now().toLocaleString();
+    this.time = currentStartTime;
+    this.displayScoreboard();
+    window.requestAnimationFrame(this.main.bind(this));
+  }
+
+  saveScore(username, points, time) {
+    const scores = JSON.parse(localStorage.getItem('scores')) || [];
+    scores.push({ username, points, time });
+    localStorage.setItem('scores', JSON.stringify(scores));
+  }
+
+  getScores() {
```

Comment:
you can separate this in separate classes that is will interact with local storage